tests/components/test_file_watcher.py

```diff
@@ -0,0 +1,59 @@
+"""The tests for the folder_watcher component."""
+import unittest
+import os
+
+from homeassistant.components.folder_watcher import (
+    DOMAIN, CONF_FOLDER, CONF_WATCHERS)
+from homeassistant.setup import setup_component
+from tests.common import get_test_home_assistant
+
+
+CWD = os.path.join(os.path.dirname(__file__))
+TEST_FOLDER = 'test_folder'
+TEST_DIR = os.path.join(CWD, TEST_FOLDER)
+TEST_TXT = 'mock_test_folder.txt'
+TEST_FILE = os.path.join(TEST_DIR, TEST_TXT)
+
+
+def create_file(path):
+    """Create a test file."""
+    with open(path, 'w') as test_file:
+        test_file.write("test")
```

Comment:
I don't really like tests that do actual I/O, especially when it's not too important to do that. This could end up being one of those pesky flakey tests. From what I see, `watchdog` itself is well tested and so there's no need for us to test `watchdog` in  Home Assistant too. We should only test whether the callbacks we *receive* from `watchdog` are properly handled by *our* code and that our setup methods call `watchdog` methods correctly. 

Maybe have a look at the mocking framework for how to do this best: https://docs.python.org/3/library/unittest.mock.html

Reply:
I copied this from the [folder sensor test](https://github.com/home-assistant/home-assistant/blob/dev/tests/components/sensor/test_folder.py), but agree this is covered by `watchdog` testing and could be omitted.